book/src/debugging.md

```diff
@@ -0,0 +1,127 @@
+# Debugging
+
+Gain deeper insights into your code's behavior with this debugging section. Here you can discover techniques and tools to explore and understand your project's internals.
+
+## Visualizing Logos Graph 
+
+Logos works by creating a graph that gets derived from the tokens that you create.  This graph visualizes how the lexer moves through different states when processing input.
+
+It may be beneficial during debugging to be able to visualize how Logos has generated its graph. 
```

Comment:
```suggestion
Hence, it may be beneficial during debugging to be able to visualize this graph,
to understand how Logos will match the various tokens. 
```